### PR TITLE
add secretsmanager:GetRandomPassword

### PR DESCRIPTION
**Pull request by markuman**
Required by https://github.com/ansible-collections/community.aws/pull/1997

### CODE REVIEW

**Comment by markuman** [comment]
oh damn

```
TASK [Fail if policy is too large] *********************************************
fatal: [localhost]: FAILED! => {
    "failed_when_result": true,
    "oversized_policies": [
        "/home/vsts/work/1/s/hacking/aws_config/../../aws/policy/security-services.yaml"
    ]
}
```

File: aws/policy/security-services.yaml [hunk 1]
```diff
@@ -106,6 +106,7 @@ Statement:
       - secretsmanager:PutResourcePolicy
       - secretsmanager:DeleteResourcePolicy
       - secretsmanager:RemoveRegionsFromReplication
+      - secretsmanager:GetRandomPassword
```
Comment (gravesm) [inline aws/policy/security-services.yaml:109]: I think the problem is that this policy statement is resource restricted and the `GetRandomPassword` permission does not operate on a secret. I just tested by adding this permission to the `AllowGlobalUnrestrictedResourceActionsWhichIncurNoFees` policy statement and the associated PR passes. Unfortunately, this policy is right at the maximum allowed characters and there's not a lot here that can be collapsed. About the only thing I can see would be to replace `kms:DisableKey` and `kms:DisableKeyRotation` with `kms:Disable*`. There are only two `kms:Disable*` permissions so this won't add anything new.

Comment (markuman) [reply aws/policy/security-services.yaml:108]: Sounds good. I've reduced the two rules to `kms:Disable*`.

File: aws/policy/security-services.yaml [hunk 2]
```diff
@@ -106,6 +105,7 @@ Statement:
...
```
[the remark this answers is not in the document]
Comment (markuman) [reply aws/policy/security-services.yaml]: done

File: aws/policy/security-services.yaml [hunk 3]
```diff
@@ -89,6 +88,7 @@ Statement:
       - logs:List*
       - secretsmanager:Describe*
       - secretsmanager:List*
+      - secretsmanager:GetRandomPassword
```
Comment (gravesm) [inline aws/policy/security-services.yaml:91]: Small nit, but could you make sure this is ordered alphabetically?

Comment (markuman) [reply aws/policy/security-services.yaml:90]: sure.

**Review by gravesm — APPROVED** [review]

**Pull request merged**